rs/rust_canisters/dfn_core/src/api.rs

```diff
@@ -876,7 +876,9 @@ pub fn canister_version() -> u64 {
 }
 
 pub fn mint_cycles(amount: u64) -> u64 {
-    unsafe { ic0::mint_cycles(amount) }
+    let mut buf = [0u8; 16];
+    unsafe { ic0::mint_cycles128(0, amount, buf.as_mut_ptr() as u32) }
+    u128::from_le_bytes(buf) as u64
```

Comment:
No need to keep anything here imo, we can remove the wrappers completely. There should be no on using a wrapper of the mint cycles API (either new or old) as the only user (CMC) is using the API directly (as it should to avoid relying on the deprecated `dfn_core`).